### PR TITLE
docs: Add solution templates debugging steps

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): https://github.com/unoplatform/uno/issues/5988

## PR Type

What kind of change does this PR introduce?
- Documentation content changes

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/articles/toc.yml [hunk 1]
```diff
@@ -428,6 +428,8 @@
           href: uno-development\creating-runtime-tests.md
         - name: UI tests with Uno.UITest
           href: uno-development\creating-ui-tests.md
+    - name: Debugging Solution Templates
+      href: uno-development\debugging-templates.md
     - name: Conventional Commits format
       href: uno-development\git-conventional-commits.md
```
Comment (carldebilly) [inline doc/articles/toc.yml:434]: Is there a logical ordering in this file?

**Review by carldebilly — APPROVED** [review]

File: doc/articles/toc.yml (hunk 1 above)
Comment (davidjohnoliver) [reply doc/articles/toc.yml:434]: it's grouped into topics, and roughly ordered by importance within each topic (or alphabetically for a few topics eg WinRT features)

File: doc/articles/uno-development/debugging-templates.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Some of the templates are reusing the same sources. For example the full solution template is defined in the vsix project, and building the `dotnet new` template creates a transformed version of the vsix template.
+
+## VSIX Templates
+
+VSIX templates are using the Visual Studio extensibility model, and run with a specific project type.
+
+To make modifications to those templates and test the result:
+- In Visual Studio, open the solution using the [Uno.UI-Wasm-hotreload-vsix-only.slnf filters](building-uno-ui.md)
+- Right click on the `UnoSolutionTemplate.VISX` project then debug
+- A new isolated testing instance of visual studio will open, allowing you to create projects using modified templatest
```
Comment (davidjohnoliver) [inline doc/articles/uno-development/debugging-templates.md:16]:
```suggestion
- A new isolated testing instance of visual studio will open, allowing you to create projects using modified templates
```

File: doc/articles/uno-development/debugging-templates.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+## dotnet new templates
+
+`dotnet new` new templates use the dotnet/templating engine, and are located in the `src/SolutionTemplate/Uno.ProjectTemplates.Dotnet` folder.
+
+To make modifications to those templates and test the result:
+- In Visual Studio, open the solution using the [Skia or WebAssembly filters](building-uno-ui.md)
+- Right click on the `Uno.ProjectTemplates.Dotnet` project, select `Pack`
+- This will create a `Uno.ProjectTemplates.Dotnet.255.255.255.255.nupkg` file
+- Install this file using `dotnet new -i C:\[your_path]\bin\Debug\Uno.ProjectTemplates.Dotnet.255.255.255.255.nupkg`
+- In another temporary folder, create a new application using the update templates with `dotnet new unoapp -o MyTestUnoApp01`
```
Comment (davidjohnoliver) [inline doc/articles/uno-development/debugging-templates.md:27]:
```suggestion
- In another temporary folder, create a new application using the updated templates with `dotnet new unoapp -o MyTestUnoApp01`
```

File: doc/articles/uno-development/debugging-templates.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Ensure that the application runs properly by building it in Visual Studio and running it on all targets.
```
Comment (davidjohnoliver) [inline doc/articles/uno-development/debugging-templates.md:29]: One might want to revert to the published dotnet new template after testing, could you add a step for that?

**Review by davidjohnoliver — APPROVED** [review]

**Review by ajpinedam — APPROVED** [review]

**Pull request merged**